File: 1394-find-lucky-integer-in-an-array/1394-find-lucky-integer-in-an-array.cs
public class Solution
{
  public int FindLucky(int[] arr)
  {
    Dictionary<int, int> freq = new();
    foreach (int num in arr)
    {
      freq[num] = freq.GetValueOrDefault(num, 0) + 1;
    }

    int result = -1;
    foreach (KeyValuePair<int, int> pair in freq)
    {
      if (pair.Key == pair.Value)
      {
        result = Math.Max(result, pair.Key);
      }
    }
    return result;
  }
}

File: 1394-find-lucky-integer-in-an-array/1394-find-lucky-integer-in-an-array.go
func findLucky(arr []int) int {
	freq := make(map[int]int)

	for _, v := range arr {
		freq[v] += 1
	}

	var result float64 = -1

	for key, freq := range freq {
		if key == freq {
			result = math.Max(float64(key), result)
		}
	}

	return int(result)
}